contracts/user/ConsumerWrapper.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import {OwnableUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {UUPSUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {BasicRandcastConsumerBase} from "./BasicRandcastConsumerBase.sol";
+import {RequestIdBase} from "../utils/RequestIdBase.sol";
+import {IAdapter} from "../interfaces/IAdapter.sol";
+// solhint-disable-next-line no-global-import
+
+contract ConsumerWrapper is RequestIdBase, BasicRandcastConsumerBase, UUPSUpgradeable, OwnableUpgradeable {
+
+    uint32 private constant _FEE_OVERHEAD = 700000;
+    uint32 private constant _GROUP_SIZE = 3;
+    uint32 private constant _MAX_GAS_LIMIT = 2000000;
```

Comment:
Let's get this from `IAdapter(adapter).getAdapterConfig();` per request. And we also need to update `GeneralRandcastConsumerBase` as well.